source/ota.c

```diff
@@ -1941,65 +1930,59 @@ static OtaJobParseErr_t handleCustomJob( const char * pJson,
     jobDoc.parseErr = OtaJobParseErrUnknown;
 
     /* If this is a valid custom job, extract job document and job ID data from the JSON payload.*/
-    if( JSONSuccess != JSON_SearchConst( pJson,
-                                         messageLength,
-                                         OTA_JSON_JOB_DOC_KEY,
-                                         strlen( OTA_JSON_JOB_DOC_KEY ),
-                                         ( const char ** ) &jobDoc.pJobDocJson,
-                                         &jobDoc.jobDocLength,
-                                         NULL ) )
-    {
-        if( JSONSuccess == JSON_SearchConst( pJson,
-                                             messageLength,
-                                             OTA_JSON_JOB_ID_KEY,
-                                             strlen( OTA_JSON_JOB_ID_KEY ),
-                                             ( const char ** ) &jobDoc.pJobId,
-                                             &jobDoc.jobIdLength,
-                                             NULL ) )
-        {
-            /* We have an unknown job parser error. Check to see if we can pass control
-             * to a callback for parsing */
-            otaAgent.OtaAppCallback( OtaJobEventParseCustomJob, &jobDoc );
+    if( ( JSONSuccess == JSON_SearchConst( pJson,
+                                           messageLength,
+                                           OTA_JSON_JOB_DOC_KEY,
+                                           strlen( OTA_JSON_JOB_DOC_KEY ),
+                                           ( const char ** ) &jobDoc.pJobDocJson,
+                                           &jobDoc.jobDocLength,
+                                           NULL ) ) &&
+
+        ( JSONSuccess == JSON_SearchConst( pJson,
+                                           messageLength,
+                                           OTA_JSON_JOB_ID_KEY,
+                                           strlen( OTA_JSON_JOB_ID_KEY ),
+                                           ( const char ** ) &jobDoc.pJobId,
+                                           &jobDoc.jobIdLength,
+                                           NULL ) ) )
+    {
+        /* We have an unknown job parser error. Check to see if we can pass control
+         * to a callback for parsing */
+        otaAgent.OtaAppCallback( OtaJobEventParseCustomJob, &jobDoc );
+
+        if( jobDoc.parseErr == OtaJobParseErrNone )
+        {
+            ( void ) memcpy( otaAgent.pActiveJobName, jobDoc.pJobId, jobDoc.jobIdLength );
```

Comment:
2 questions:

1. Are we sure that `otaAgent.pActiveJobName` has at least `jobDoc.jobIdLength` bytes of space?  It looks like `otaAgent.pActiveJobName` is `OTA_JOB_ID_MAX_SIZE` bytes based on line 1971.
1. Why are we overwriting the job name with the job ID here?

Some comments explaining this could be helpful.

Reply:
Regarding 2., it is just copying the job ID (that is parsed from the incoming PUBLISH payload buffer) to the `otaAgent` object so that a status update can be sent for the Job to the service through the `otaControlInterface.updateJobStatus` function.

Reply:
Added check in fd787aa3109f0ec2a55a00771bb56738580a24a3